### PR TITLE
Swift5 compatibility

### PR DESCRIPTION
**Pull request by ra1028**
- Fix performance degradation in #55
- Remove redundant Swift4.2 compatibility
- Migrate project with Xcode5.0
- `StagedChangeset` conforms to `MutableCollection` again
- Update travis CI settings, Gems, README and Docs

CC: @alanzeino @Kaspik @BjornRuud

### CODE REVIEW

File: DifferenceKit.podspec [hunk 1]
```diff
@@ -13,6 +13,7 @@ Pod::Spec.new do |spec|
   spec.license = { :type => 'MIT', :file => 'LICENSE' }
   spec.requires_arc = true
   spec.default_subspecs = 'Core', 'UIKitExtension'
+  spec.swift_version = "4.2"
```
Comment (Kaspik) [inline DifferenceKit.podspec]: Shouldn't this be 5.0?

Comment (ra1028) [reply DifferenceKit.podspec]: 👌

Comment (Kaspik) [reply DifferenceKit.podspec]: Probably shouldn't 😂

Comment (ra1028) [reply DifferenceKit.podspec]: @Kaspik 
DifferenceKit should supports both Swift4.2 and 5.0.
 Xcode10.2 is able to build sources that are written in either Swift 5, Swift 4.2.

**Comment by alanzeino** [comment]
👍

**Comment by Z-JaDe** [comment]
spec.swift_versions = ["4.2", "5.0"]

**Pull request merged**